in-person-payments-example/Controllers/ApiController.cs

```diff
@@ -0,0 +1,273 @@
+                        {
+                            Result = "success"
+                        });
+                    case ResultType.Failure:
+                        table.PaymentStatus = PaymentStatus.RefundFailed;
+                        return Ok(new CreateReversalResponse()
+                        {
+                            Result = "failure",
+                            RefusalReason = "Payment terminal responded with: " + HttpUtility.UrlDecode(reversalResponse.Response.AdditionalResponse)
+                        });
+                    case ResultType.Partial:
+                        throw new NotImplementedException(nameof(ResultType.Partial));
+                    default:
+                        return BadRequest(new CreateReversalResponse()
+                        {
+                            Result = "failure",
+                            RefusalReason = _poiId == null ? "Could not reach payment terminal - POI ID is not set" : $"Could not reach payment terminal with POI ID {_poiId}"
+                        });
+                }
+            }
+            catch (HttpClientException e)
+            {
+                _logger.LogError(e.ToString());
+                table.PaymentStatus = PaymentStatus.NotPaid;
+                return StatusCode(e.Code, new CreateReversalResponse()
+                {
+                    Result = "failure",
+                    RefusalReason = $"ErrorCode: {e.Code}, see logs"
+                });
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e.ToString());
+                throw;
+            }
+        }
+
+        [HttpPost("api/abort/{tableName}")]
+        public async Task<ActionResult<SaleToPOIResponse>> Abort(string tableName, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                TableModel table = _tableService.Tables.FirstOrDefault(t => t.TableName == tableName);
+
+                if (table?.PaymentStatusDetails?.ServiceId == null)
+                {
+                    return NotFound();
+                }
+
+                SaleToPOIResponse abortResponse = await _posAbortService.SendAbortRequestAsync(table.PaymentStatusDetails.ServiceId, _poiId, _saleId, cancellationToken);
+                table.PaymentStatus = PaymentStatus.NotPaid;
+                return Ok(abortResponse);
+            }
+            catch (HttpClientException e)
+            {
+                _logger.LogError(e.ToString());
+                return StatusCode(e.Code, new CreateReversalResponse()
+                {
+                    Result = "failure",
+                    RefusalReason = $"ErrorCode: {e.Code}, see logs"
+                });
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e.ToString());
+                throw;
+            }
+        }
+
+        [HttpGet("api/get-transaction-status/{tableName}")]
```

Comment:
`api/get-transaction-status` is not used I think. Leave but add a comment to describe how it can be used?

Reply:
I'll add a button, I forgot to add this as I just visited the URL to get the response